dashboard-server/queries/responses/response.test.js

```diff
@@ -57,33 +71,188 @@ describe('Response queries', () => {
       responses[0].last_response.should.equal('{ "text": "last" }');
-      responses[0].surveyid.should.equal(
-        survey2.id,
-      );
+      responses[0].surveyid.should.equal(survey2.id);
       responses[1].first_response.should.equal('{ "text": "first" }');
       responses[1].last_response.should.equal('{ "text": "last" }');
-      responses[1].surveyid.should.equal(
-        survey.id,
+      responses[1].surveyid.should.equal(survey.id);
+    });
+  });
+
+  describe('.all()', () => {
+    it('should return all responses for a survey created by a user', async () => {
+      const user = {
+        email: 'test3@vlab.com',
+      };
+      const newUser = await User.create(user);
+
+      const survey = await Survey.create({
+        created: new Date(),
+        formid: 'biy23',
+        form: '{"form": "form detail"}',
+        messages: '{"foo": "bar"}',
+        shortcode: 231,
+        userid: newUser.id,
+        title: 'Survey',
+        metadata: '{}',
+        survey_name: 'Survey123',
+        translation_conf: '{}',
+      });
+
+      const survey2 = await Survey.create({
+        created: new Date(),
+        formid: '3hu23',
+        form: '{"form": "form detail"}',
+        messages: '{"foo": "bar"}',
+        shortcode: 123,
+        userid: newUser.id,
+        title: 'Other survey',
+        metadata: '{}',
+        survey_name: 'Survey321',
+        translation_conf: '{}',
+      });
+
+      const timestamp = '2022-06-06 10:00:00+00:00';
+      const timestamp2 = '2022-06-06 10:00:00+00:01';
+      const timestamp3 = '2022-06-06 10:00:00+00:02';
+
+      const MOCK_QUERY = `INSERT INTO responses(parent_surveyid, parent_shortcode, surveyid, shortcode, flowid, userid, question_ref, question_idx, question_text, response, seed, timestamp)
+      VALUES
+        ('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100001, '126', 'ref', 10, 'text', '{ "text": "last" }', '6789', '${timestamp}')
+       ,('${survey2.id}', '202', '${
+        survey2.id
+      }', '202', 100003, '126', 'ref', 10, 'text', '{ "text": "do not return me" }', '6789', '${timestamp2}')
+       ,('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100004, '126', 'ref', 10, 'text', '{ "text": "first" }', '6789', '${timestamp3}')`;
+
+      await vlabPool.query(MOCK_QUERY);
+
+      const responses = await Response.all({
+        email: user.email,
+        survey: survey.survey_name,
+      });
+
+      responses.should.equal([
+        {
+          parent_surveyid: `${survey.id}`,
+          parent_shortcode: '101',
+          surveyid: `${survey.id}`,
+          flowid: '100004',
+          userid: '126',
+          question_ref: 'ref',
+          question_idx: '10',
+          question_text: 'text',
+          response: '{ "text": "first" }',
+          timestamp: `${timestamp3}`,
+          metadata: null,
+          pageid: null,
+          translated_response: null,
+        },
```

Comment:
Is this the response I can base of? @sssoja

Reply:
Yes, this is the structure of the JSON object returned by the `Response.all` query. 

An example response would like this:

```
[
  {
    parent_surveyid: 'be5ae9dd-0189-478e-8a3d-4d8ead8240a4',
    parent_shortcode: '101',
    surveyid: 'be5ae9dd-0189-478e-8a3d-4d8ead8240a4',
    flowid: '100004',
    userid: '126',
    question_ref: 'ref',
    question_idx: '10',
    question_text: 'text',
    response: '{ "text": "first" }',
    timestamp: '2022-06-06 09:58:00+00:00',
    metadata: null,
    pageid: null,
    translated_response: null
  },
  {
    parent_surveyid: 'be5ae9dd-0189-478e-8a3d-4d8ead8240a4',
    parent_shortcode: '101',
    surveyid: 'be5ae9dd-0189-478e-8a3d-4d8ead8240a4',
    flowid: '100001',
    userid: '126',
    question_ref: 'ref',
    question_idx: '10',
    question_text: 'text',
    response: '{ "text": "last" }',
    timestamp: '2022-06-06 10:00:00+00:00',
    metadata: null,
    pageid: null,
    translated_response: null
  }
]
```